aml/src/resource.rs

```diff
@@ -1,112 +1,293 @@
+}
+
+
+#[derive(Debug, PartialEq)]
+pub struct AddressSpaceDescriptor {
+    resource_type: AddressSpaceResourceType,
+    is_maximum_address_fixed: bool,
+    is_minimum_address_fixed: bool,
+    decode_type: AddressSpaceDecodeType,
+
+    granularity: u64,
+    address_range: (u64, u64),
+}
+
+#[derive(Debug, PartialEq)]
+pub enum MemoryRangeDescriptor {
+    FixedLocation { 
+        is_writable: bool,
+        base_address: u32,
+        range_length: u32
+    }
+}
+
+fn fixed_memory_descriptor(bytes: &[u8]) -> Result<Resource, AmlError> {
+    /*
+     * -- 32-bit Fixed Memory Descriptor ---
+     * Offset     Field Name                              Definition      
+     * Byte 0     32-bit Fixed Memory Range Descriptor    Value = 0x86 (10000110B) – Type = 1, Large item name = 0x06
+     * Byte 1     Length, bits [7:0]                      Value = 0x09 (9)
+     * Byte 2     Length, bits [15:8]                     Value = 0x00
+     * Byte 3     Information                             This field provides extra information about this memory. 
+     *                                                    Bit [7:1]   Ignored
+     *                                                    Bit [0]     Write status, _RW
+     *                                                        1  writeable (read/write)
+     *                                                        0  non-writeable (read-only))
+     * Byte 4     Range base address, _BAS bits [7:0]     Address bits [7:0] of the base memory address for which the card may be configured. 
+     * Byte 5     Range base address, _BASbits [15:8]     Address bits [15:8] of the base memory address for which the card may be configured.
+     * Byte 6     Range base address, _BASbits [23:16]    Address bits [23:16] of the base memory address for which the card may be configured. 
+     * Byte 7     Range base address, _BASbits [31:24]    Address bits [31:24] of the base memory address for which the card may be configured.
+     * Byte 8     Range length, _LEN bits [7:0]           This field contains bits [7:0] of the memory range length. The range length provides the length of the memory range in 1-byte blocks.
+     * Byte 9     Range length, _LEN bits[15:8]           This field contains bits [15:8] of the memory range length. The range length provides the length of the memory range in 1-byte blocks.
+     * Byte 10    Range length, _LEN bits [23:16]         This field contains bits [23:16] of the memory range length. The range length provides the length of the memory range in 1-byte blocks.
+     * Byte 11    Range length, _LEN bits [31:24]         This field contains bits [31:24] of the memory range length. The range length provides the length of the memory range in 1-byte blocks.
+     */
+    if bytes.len() < 12 {
+        return Err(AmlError::ResourceDescriptorLengthIncorrect);
+    }
+
+    let information = bytes[3];
+    let is_writable = information.get_bit(0);
+
+    let base_address = 
```

Comment:
This and `range_length` can probably be `LittleEndian::read_u32(&bytes[4..8])` and `LittleEndian::read_u32(&bytes[8..12])` respectively.